Fix positioning error

iansan5653 opened the pull request:
Fixes a bug where the menu position would be calculated incorrectly because `dom-input-range` gives absolute positions and we were still offsetting by the position of the input.

This approach should be generally more robust by re-adjusting the position if necessary after setting it.

keithamus commented:
Does this account for scroll containers? It's dropping the `scrollx` code which tells me it might not work properly if scrolled.

iansan5653 commented:
Sorry, didn't intend to mark this ready for review yet. I need to figure out a way to test this locally in dotcom. 

> Does this account for scroll containers? It's dropping the `scrollx` code which tells me it might not work properly if # scrolled.

Theoretically scrolling should work as expected: `DomInputRange` is already accounting for scroll. When `getBoundingClientRect` is called on the range, the result is _exactly_ where we want the menu to be relative to the document, so it's just a matter of making sure that's where the menu ends up. 

I am wondering if, instead of adjusting `top`/`left` twice, we should use `transform` to fix any discrepancy. That might be faster and the code would be cleaner 🤔

iansan5653 commented:
> I am wondering if, instead of adjusting `top`/`left` twice, we should use `transform` to fix any measured discrepancy. That might be faster and the code would be cleaner 🤔

Decided to go with this route since `transform` is generally faster and allows us to do all the positioning in one step. We calculate the current position and the desired position, then simply use `translate` to adjust the current position to be the desired position -- no need to worry about absolute/relative/fixed/top-layer positioning.

iansan5653 commented:
Okay, now we're ready for review. Tested in dotcom by manually copying in the changes and it's working great. I updated the approach to only have to set `top` and `left` once (fewer repaints!) and extracted a `positionMenu` function.

keithamus reviewed (approved):
👍 LGTM

silverwind commented:
Also noticed that the positioning in `@github/text-expander-element@2.7.0` is rather broken compared to 2.6.0, e.g. too far in both the x and y axis. I hope this fixes that.

<img width="337" alt="image" src="https://github.com/github/text-expander-element/assets/115237/7e70e675-6e7e-43e3-8f72-b657f9db3741">

silverwind commented:
Thanks, I can confirm 2.7.1 fixed that issue for me.
